torch/csrc/quantized/quantized_backward.cpp

```diff
@@ -0,0 +1,76 @@
+#include <ATen/native/quantized/PackedParams.h>
+#include <torch/library.h>
+#include <torch/torch.h>
+
+namespace {
+using namespace torch::autograd;
+using namespace at;
+
+class PackedLinearWeightDynamicBackward :
+public Function<PackedLinearWeightDynamicBackward> {
+ public:
+  // Note that both forward and backward are static functions
+
+  // bias is an optional argument
+  static torch::Tensor forward(
+    AutogradContext *ctx,
+    at::Tensor input,
+    const c10::intrusive_ptr<LinearPackedParamsBase>& packed_weight,
+    bool reduce_range) {
+
+    static auto op =
+        at::Dispatcher::singleton()
+            .findSchemaOrThrow("quantized::linear_dynamic", "")
+            .typed<at::Tensor(
+              at::Tensor,
+              c10::intrusive_ptr<LinearPackedParamsBase,
+              c10::detail::intrusive_target_default_null_type<LinearPackedParamsBase>> const&,
+              bool)>();
+    auto output = op.redispatch(
+      DispatchKeySet({DispatchKey::CPU}), input, packed_weight, reduce_range);
+    auto [weight, bias] = packed_weight->unpack();
+    TORCH_CHECK(weight.qscheme() == c10::kPerTensorAffine);
+    // TO-DO : passing a bias does not work, hence create an arbitrary one in backward loop
```

Comment:
If you need to save non-Tensor data, you can do so by doing `ctx->saved_data["foo"] = foo`

Reply:
@albanD bias is also a tensor.

@kwanghoon-meta note that you are unpacking weights at forward, so how you will have two copies of weights. One in the packed_weight another for gradient that you are saving. So this will increase your memory footprint if that is what you wanted to minimize. Really what you want is to save packed_weight but I dont know if you can do that with autograd. But if this memory increase is fine then this works.

Reply:
I think the reason passing bias dose not work is because gradient uses transposed weights where as bias shape is aligned for non-tranposed weight. But thinking more why do you need to pass bias at all or even use bias in the backward since gradient of bias addition w.r.t input is really zero, no?

Reply:
> > Really what you want is to save packed_weight but I dont know if you can do that with autograd.
> 
> Just use `saved_data` ;)

oh thats right. You just said that above.